### PR TITLE
Project: Weather App

### PR DESCRIPTION
**Pull request by brenoxav**
- Use of OpenWeatherMap API;
- Asynchronous Javascript;
- Input field to search by city name;
- Use current geolocation to search local weather;
- Switch between Celsius and Fahrenheit;
- Display different icons according to the weather condition and time of the day;
- Animated loading element.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+  "name": "weather-app",
+  "description": "![](https://img.shields.io/badge/Microverse-blueviolet)",
+  "version": "1.0.0",
+  "private": true,
+  "main": "index.js",
+  "scripts": {
+    "dev": "webpack --watch --mode development",
+    "build": "webpack --mode production",
+    "start": "webpack-dev-server --mode development --open"
+  },
```
Comment (oloomoses) [inline package.json:11]: - [x] Kindly install webpack-dev-server in devDependencies for the `npm start` command to execute.

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+const API_KEY = 'e24777d55aaff3d179ae147424695333';
+
+function convertToUnit(tempC) {
+  if (unitCheckbox.checked) {
+    temperatureUnit.forEach((unit) => { unit.textContent = '°C'; });
+    return tempC;
+  }
+  temperatureUnit.forEach((unit) => { unit.textContent = '°F'; });
+  return (((tempC * 9) / 5) + 32);
+}
```
Comment (oloomoses) [inline src/index.js]: - [x] Use arrow function instead of the function word

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+function renderWeather(data) {
+  locationName.textContent = `${data.name}, ${data.sys.country}`;
+  weatherIcon.style.backgroundImage = `url('${ICON_URL}${data.weather[0].icon}.svg')`;
+  weatherDescription.textContent = data.weather[0].description;
+  temperatureValue.textContent = `${Math.round(convertToUnit(data.main.temp))}`;
+  minTemperature.textContent = `${Math.round(convertToUnit(data.main.temp_min))}`;
+  maxTemperature.textContent = `${Math.round(convertToUnit(data.main.temp_max))}`;
+}
```
Comment (oloomoses) [inline src/index.js]: - [x] Use arrow function instead of the function word

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    })
+    .catch(() => {
+      backdrop.classList.add('display');
+      warningMessage.innerHTML = `
+      <p class="warning-message">LOCATION NOT FOUND</p>
+      </br>
+      <p class="warning-message">Click the pin icon above to use you current location OR type a city name in the search field.</p>
+      `;
+    });
+}
```
Comment (oloomoses) [inline src/index.js]: - [x] Use arrow function instead of the function word

**Review by oloomoses — APPROVED** [review]
Hello @brenoxav, 
You have done a great job on this weather app :muscle: 
## Highlights
- I am able to search for a location and get weather data :100: 
- I am able to toggle between Celsius and Fahrenheit :+1: 
- Well done on the linter configuration
- Great job on the readme :clap: 

## To Improve
**However, some changes need to be done before this project can be approved :recycle:** 

- [x] Kindly ensure that the app changes its appearance based on the data received ([As per the requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png))
- [x] Kindly ensure that webpack-dev-server is installed in the devDepencies for the `npm start` command to execute. Check the comments for more.
- [x] Kindly use arrow functions instead of the function keyword. Check the comments for more.
- [x] The link to the live demo provided in the readme file does not work. Kindly update it with the working link

## Optional
- [ ] Kindly move your logic to a different file so as to separate them from the DOM syntax. By logic, I refer to any code that does not interact directly with the DOM.

**Kindly make the above changes before requesting another review**

Feel free to reach me on slack or leave a comment while mentioning me and I will  be happy to respond

_Slack: Oloo Moses_

**Comment by brenoxav** [comment]
The required changes were done.

**Review by oloomoses — APPROVED** [review]
Hello @brenoxav, 
## Congratulation :muscle: :tada: 
### Highlights
- Use of es6 arrow functions :heavy_check_mark:
- The app changes its appearance based on the data received :heavy_check_mark:
- Webpack-dev-server is installed in the devDepencies :heavy_check_mark:
- Working live demo link in the readme :heavy_check_mark:

**Well done on implementing the changes suggested by previous reviewers  :muscle:** 
### This project meets the minimum requirements and therefore Approved :clap: 

**I wish best of luck in your next project and see you at the top :rocket:** 

_Slack: Oloo Moses_

**Pull request merged**